### ISSUE TITLE
fix board lookup for layer names with `/`

### ISSUE THREAD

**Issue by gavin-ts** [terrastruct/d2#1727]
```d2
"x.a" -> "y / b"
```

object ids can contain `/` so board lookup needs to account for this, passing the path as `[]string` would be better.

If `/` is a valid character in layer names, wouldn't it also break this existing code?

```go
// boardPath comes in the form of "x/layers/z/scenarios/a"
// or "layers/z/scenarios/a"
// or "x/z/a"
func (d *Diagram) GetBoard(boardPath string) *Diagram {
	path := strings.Split(boardPath, string(os.PathSeparator))
	if len(path) == 0 || len(boardPath) == 0 {
		return d
	}
```

If so, it seems like a bug.

_Originally posted by @landmaj in https://github.com/terrastruct/d2/pull/1725#discussion_r1391620205_

**Comment by landmaj** [comment]
SVG directory structure is also broken when layer names contain `/`.

```d2
a -> b

layers: {
  x / y: {
    c -> d
  }
}
```

```bash
❯ d2 index.d2
success: successfully compiled index.d2 to index/x / y.svg in 78.016917ms
success: successfully compiled index.d2 to index/index.svg in 76.312875ms
```

This results in following directory tree:
```
index
├── x 
│   └──  y.svg
└── index.svg
```

**Comment by landmaj** [comment]
I fixed this in #1725
